examples/transmission_main_example.py

```diff
@@ -77,10 +61,63 @@ def main(args):
     axis('off')
     show()
 
+
+def main(args):
+    input_filename = str(args.filename)
+    split_filename = input_filename.split(".")
+    json_filename = split_filename[0]+'.json'
+    try:
+        assert split_filename[1] in ('json','xls','csv','xlsm')
+    except AssertionError as e:
+        print(f'invalid file extension .{split_filename[1]}')
+        raise e
+    if split_filename[1] != 'json':
+        print(f'parse excel input to {json_filename}')
+        convert_file(input_filename)
+
+    json_data = load_json(json_filename)
+    read_eqpt_library(EQPT_LIBRARY_FILENAME)
+
+    network = network_from_json(json_data)
+    build_network(network)
+
+    spacing = 0.05 #THz
+    si = SpectralInformation() # !! SI units W, Hz
+    si = si.update(carriers=tuple(Channel(f, (191.3+spacing*f)*1e12, 
+            32e9, 0.15, Power(1e-3, 0, 0)) for f in range(1,97)))
+
+    trx = [n for n in network.nodes() if isinstance(n, Transceiver)]
+    if args.list>=1:
+        print(*[el.uid for el in trx], sep='\n')
```

Comment:
You dont account for the possibility that the user did not supply a source and sink agument, other than by specifying an invalid default value `""`.

Either:
- make these parameters completely optional and dynamically determine a reasonable default value (which is not particularly meaningful,)
- make these values mandatory.

Reply:
Additionally, you also don't account for the possibility that the source and sink are invalid.

For now, it's okay that you're assuming that the node uids encode the network element type in the first four characters, but that may change later (in which case we'll have to fix this code.)

I would break this into two separate loops which `break` once they have identified the correct element, plus a check at the end.

You could write:
```
source = None
for el in trx:
    if args.source == el.uid[4:]:
        source = el
        break
if source is None:
    raise ValueError(f"invalid source node specified: {args.source!r}")
```

You could simplify to:
```
source = next((el for el in trx if args.source == el.uid[4:]), None)
if source is None:
    raise ValueError(f"invalid source node specified: {args.source!r}")
```

Or even:
```
try:
    source = next(el for el in trx if args.source == el.uid[4:])
except StopIteration as e:
    raise ValueError(f"invalid source node specified: {args.source!r}") from e
```

Reply:
Also not finding this useful,
the code you put is from previous PR
the new code accounts for missing or wrong value by providing a default node[0] and node(1] source and sink. (Like the previous PR did by the way...). And logging (printing) to the user that default values were used. These default values are fine for an example to the user and have the advantage to always exist because they are the 2 1st nodes of the network topology.